### PR TITLE
fix: Fix deconv kernel channel num_output_maps where wts are ITensor

### PR DESCRIPTION
**Pull request by andi4191**
# Description

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.
In current implementation deconv layer picks up the in_channel for kernel incorrectly when weight is also an ITensor.
This PR fixes that issue.

[Conv2d](https://pytorch.org/docs/stable/generated/torch.nn.Conv2d.html)
[ConvTranspose2d](https://pytorch.org/docs/stable/generated/torch.nn.ConvTranspose2d.html)

For conv2d the kernel dimensions are: [out_channels, in_channels / group, ... ]
For convTranspose2d the kernel dimension should be: [in_channel, out_channels / group, ... ]

Fixes # (issue)

## Type of change

Please delete options that are not relevant and/or add your own.

- Bug fix (non-breaking change which fixes an issue)
- New feature (non-breaking change which adds functionality)
- Breaking change (fix or feature that would cause existing functionality to not work as expected)
- This change requires a documentation update

# Checklist:

- [ ] My code follows the style guidelines of this project (You can use the linters)
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas and hacks
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests to verify my fix or my feature
- [ ] New and existing unit tests pass locally with my changes
- [ ] I have added the relevant labels to my PR in so that relevant reviewers are notified

### CODE REVIEW

**Comment by andi4191** [comment]
Hi @bowang007 

Request you to please review this PR for merge?

**Review by peri044 — APPROVED** [review]
Thanks. LGTM. I verified the test passes on local
